### PR TITLE
Adds visible ripples indicating shuttle's transit

### LINKED ISSUE #12168: White Ship can kill (ramming) remotely (EDIT: Shuttle hitboxes are squared?)
Just had an instance where, floating through space, I saw the White Ship's outer hull on the very edge of my vision, then started floating in the other direction, and was dead.

According to NikNakFlak, I had 700 brute, which is consistent with being rammed by a teleporting white ship. What isn't consistent is that the ramming didn't occur. It appeared ahead of me, oh yes, but I was barely close enough to see it, let alone be rammed by it. 

Yet another glitch-caused death to add onto the conga line of buggy premature game enders which I have endured over the last couple of months. So, what could have caused this one?

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
rscadd: Due to budget cuts, the shuttle's hyperspace engines now create a visual distortion at their destination, a few seconds before arrival. Crewmembers are encouraged to use these "ripples" as an indication of where not to stand.
fix: Shuttles now only run over mobs in turfs they are moving into, rather than the entire rectangle.
/:cl:
- Adds a new layer RIPPLE_LAYER, above vines, below ghosts
- Ripples appear at 10 seconds (the engines online time) and fade in
  from fully transparent to fully visible during that time.
- Here is a video of the effect. https://youtu.be/K0sruhMtcjA
- I've tested it, and it works fine with shuttle rotation and non-EASTfacing docks.

Fixes #12168.

### CODE REVIEW

**Comment by demonfiren** [comment]
>hyperspace
>not bluespace
smh tbh fam

File: code/modules/shuttle/shuttle.dm [hunk 1]
```diff
@@ -365,6 +371,28 @@
...
+	for(var/i in turfs)
+		ripples += new /obj/effect/ripple(i)
+
+/obj/docking_port/mobile/proc/ripple_area(obj/docking_port/stationary/S1)
+	var/list/L0 = return_ordered_turfs(x, y, z, dir, areaInstance)
+	var/list/L1 = return_ordered_turfs(S1.x, S1.y, S1.z, S1.dir)
+
+	var/list/ripple_turfs = list()
+
+	for(var/i=1, i<=L0.len, ++i)
```
Comment (RemieRichards) [inline code/modules/shuttle/shuttle.dm]: ```for(var/I in 1 to L0.len)``` please

File: code/modules/shuttle/ripple.dm [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/obj/effect/ripple
+	name = "bluespace ripple"
+	desc = "Something is coming through bluespace, you can see the \
+		visual disturbances. It's probably best not to be on top of these \
+		when whatever is tunneling comes through."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "medi_holo"
```
Comment (ChangelingRain) [inline code/modules/shuttle/ripple.dm]: By the time these are visible the animation will have finished; it doesn't repeat.

**Comment by GunHog** [comment]
Gif please?

**Comment by Niknakflak** [comment]
Gif please?

**Comment by RemieRichards** [comment]
Gif please?

**Comment by MrStonedOne** [comment]
Gif please?

**Comment by Iamgoofball** [comment]
Gif please?

**Comment by Cheridan** [comment]
Gif please?

**Comment by timkoster1** [comment]
Gif Please?

File: code/modules/shuttle/ripple.dm (hunk 2 above)
Comment (coiax) [inline code/modules/shuttle/ripple.dm]: They're placeholder animations anyway, although I do like the circle that it ends on.

**Comment by PKPenguin321** [comment]
Gif please?

**Comment by coiax** [comment]
Gif please?

**Comment by WJohn** [comment]
Sprites are very poor, but I suppose you're waiting on me to do something like that, aren't you?

I can look into it, see if I can whip up anything good.

**Comment by optimumtact** [comment]
Any chance on those sprites soon or should I just go with what we have?

**Comment by Cheridan** [comment]
we can always just add the sprites later

**Pull request merged**